test/util/removeUnnecessaryParens.test.js

```diff
@@ -22,7 +22,7 @@ describe('removeUnnecessaryParens', function () {
     ['(((-5)))', '-5'],
     ['((4+5)) + ((2^3))', '(4 + 5) + 2^3'],
     ['(2x^6 + -50 x^2) - (x^4)', '2x^6 - 50x^2 - x^4'],
-    ['(x+4) - (12 + x)', 'x + 4 - (12 + x)'],
+    ['(x+4) - (12 + x)', '(x + 4) - (12 + x)'],
```

Comment:
If I flattenOperands cloned node, 22 tests fail.

```
function print(node, showPlusMinus=false) {
  // let string = printTreeTraversal(node);

  const inputNode = flattenOperands(clone(node));
  let string = printTreeTraversal(inputNode);

  if (!showPlusMinus) {
    string = string.replace(/\s*?\+\s*?\-\s*?/g, ' - ');
  }
```

Makes this tests fail. I'll remove similar, for brevity.

```
  1) Equation constructor 2*x^2 + x = 4:
     TypeError: node.clone is not a function

  4) factorQuadratic x^2 + 2x -> x(x + 2):
      AssertionError: 'x * (x + 2)' deepEqual 'x(x + 2)'
      -x * (x + 2)
      +x(x + 2)

  14) combinePolynomialTerms multiplication 2x * x^2 * 5x -> 10x^4:
      AssertionError: '10x^4' deepEqual '10 * x^4'
      -10x^4
      +10 * x^4

  17) distribute (x+2+y)*x*7 -> (x^2 + 2 * x + y * x) * 7:
      AssertionError: '(x * x + 2x + y * x) * 7' deepEqual '(x * x + 2 * x + y * x) * 7'
      -(x * x + 2x + y * x) * 7
      +(x * x + 2 * x + y * x) * 7

  19) collects and combines like terms 2x^2 * y * x * y^3 -> 2 * x^3 * y^4:
      AssertionError: '2x^3 * y^4' deepEqual '2 * x^3 * y^4'
      -2x^3 * y^4
      +2 * x^3 * y^4

  22) distribution - x*y^4 (6x * y^2 + 5x*y - 3x) -> -6 * x^2 * y^6 - 5 * x^2 * y^5 + 3 * x^2 * y^4:
      AssertionError: '-6x^2 * y^6 - 5x^2 * y^5 + 3x^2 * y^4' deepEqual '-6 * x^2 * y^6 - 5 * x^2 * y^5 + 3 * x^2 * y^4'
      --6x^2 * y^6 - 5x^2 * y^5 + 3x^2 * y^4
      +-6 * x^2 * y^6 - 5 * x^2 * y^5 + 3 * x^2 * y^4